### PR TITLE
Zone shard

### PR DESCRIPTION
**Pull request by robertlincecum**

### CODE REVIEW

File: src/constants/shards.ts [hunk 1]
```diff
@@ -1,72 +1,72 @@
...
+    Paxos2 = '0x11',
+    Paxos3 = '0x12',
+    Hydra = '0x2',
+    Hydra1 = '0x20',
+    Hydra2 = '0x21',
+    Hydra3 = '0x22',
+    Prime = '0x',
+}
+
+function fromBytes(shard: string): Shard {
```
Comment (rileystephens28) [inline src/constants/shards.ts:19]: I know that this method is not exported but I believe the function naming could be more clear by changing it to `zoneFromBytes`, `deriveZoneFromBytes` or something along those lines

File: src/constants/zones.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    Cyprus3 = '0x02',
+    Paxos1 = '0x10',
+    Paxos2 = '0x11',
+    Paxos3 = '0x12',
+    Hydra1 = '0x20',
+    Hydra2 = '0x21',
+    Hydra3 = '0x22',
+}
+
+function fromBytes(zone: string): Zone {
```
Comment (rileystephens28) [inline src/constants/zones.ts:13]: Same situation as the shard `fromBytes` method, let's rename for clarity.

File: src/providers/abstract-provider.ts [hunk 3]
```diff
@@ -425,117 +425,117 @@ export type PerformActionRequest =
...
           method: 'getBlockNumber';
-          shard?: string;
+          shard?: Shard;
       }
     | {
           method: 'getCode';
           address: string;
           blockTag: BlockTag;
-          shard: string;
+          shard: Shard;
```
Comment (rileystephens28) [inline src/providers/abstract-provider.ts:476]: This refers to getting a smart contracts bytecode which only applies to zones

File: src/providers/abstract-provider.ts [hunk 4]
```diff
@@ -425,117 +425,117 @@ export type PerformActionRequest =
...
           method: 'getGasPrice';
           txType: boolean;
-          shard?: string;
+          zone?: Zone;
       }
     | {
           method: 'getLogs';
           filter: PerformActionFilter;
-          shard: string;
+          shard: Shard;
```
Comment (rileystephens28) [inline src/providers/abstract-provider.ts:486]: Logs are created by smart contracts so this should be zone

File: src/providers/abstract-provider.ts [hunk 5]
```diff
@@ -425,117 +425,117 @@ export type PerformActionRequest =
...
-          shard?: string;
+          zone?: Zone;
       }
     | {
           method: 'getStorage';
           address: string;
           position: bigint;
           blockTag: BlockTag;
-          shard: string;
+          shard: Shard;
```
Comment (rileystephens28) [inline src/providers/abstract-provider.ts:497]: This method relates to returning smart contract data at some storage position so this should be zone

File: src/transaction/abstract-transaction.ts [hunk 6]
```diff
@@ -32,10 +33,10 @@ export interface TransactionLike {
 }
 
 /**
- * @TODO write documentation for this interface.
- *
  * @category Transaction
  * @todo Write documentation for this interface.
```
Comment (rileystephens28) [inline src/transaction/abstract-transaction.ts:37]: duplicate todo line

File: src/transaction/abstract-transaction.ts [hunk 7]
```diff
@@ -160,20 +161,20 @@ export interface ProtoTransaction {
...
```
Comment (rileystephens28) [inline src/transaction/abstract-transaction.ts:165]: duplicate todo line

File: src/transaction/abstract-transaction.ts [hunk 8]
```diff
@@ -160,20 +161,20 @@ export interface ProtoTransaction {
...
  */
 export interface ProtoAccessList {
     access_tuples: Array<ProtoAccessTuple>;
 }
 
 /**
- * @TODO write documentation for this interface.
- *
  * @category Transaction
  * @todo Write documentation for this interface.
```
Comment (rileystephens28) [inline src/transaction/abstract-transaction.ts:175]: Duplicate todo line

File: src/transaction/abstract-transaction.ts [hunk 9]
```diff
@@ -188,17 +189,16 @@ type allowedSignatureTypes = Signature | string;
  * much (the {@link ProtoTransaction.value | **value** } in ether) the operation should entail.
  *
  * @category Transaction
```
Comment (rileystephens28) [inline src/transaction/abstract-transaction.ts:191]: Duplicate category line

File: src/wallet/hdwallet.ts
Comment (rileystephens28) [inline src/wallet/hdwallet.ts:1]: This file will be replaced by @alejoacosta74 updated wallet implementation feature very soon

File: src/wallet/qi-hdwallet.ts
Comment (rileystephens28) [inline src/wallet/qi-hdwallet.ts:1]: This file will be replaced by @alejoacosta74 updated wallet implementation feature very soon

File: src/wallet/quai-hdwallet.ts
Comment (rileystephens28) [inline src/wallet/quai-hdwallet.ts:1]: This file will be replaced by @alejoacosta74 updated wallet implementation feature very soon

File: src/constants/shards.ts (hunk 1 above)
Comment (robertlincecum) [reply src/constants/shards.ts:19]: @rileystephens28 I thought about that but decided fromBytes made more sense given the context. I can change it to have the context in the name though.

File: src/providers/abstract-provider.ts (hunk 3 above)
Comment (robertlincecum) [reply src/providers/abstract-provider.ts:476]: @rileystephens28 makes sense

File: src/wallet/hdwallet.ts
Comment (robertlincecum) [reply src/wallet/hdwallet.ts:1]: file should stay there for now though so when he rebases he can see how the enums should be used

**Pull request merged**